warehouse/internal/loadfiles/loadfiles.go

```diff
@@ -0,0 +1,348 @@
+				if err != nil {
+					return fmt.Errorf("unmarshalling response from pgnotifier: %w", err)
+				}
+				if len(output) == 0 {
+					lf.Logger.Errorf("[WH]: No LoadFiles returned by wh worker")
+					continue
+				}
+				for i := range output {
+					loadFiles = append(loadFiles, model.LoadFile{
+						TableName:             output[i].TableName,
+						Location:              output[i].Location,
+						TotalRows:             output[i].TotalRows,
+						ContentLength:         output[i].ContentLength,
+						StagingFileID:         output[i].StagingFileID,
+						DestinationRevisionID: output[i].DestinationRevisionID,
+						UseRudderStorage:      output[i].UseRudderStorage,
+						SourceID:              job.Upload.SourceID,
+						DestinationID:         job.Upload.DestinationID,
+						DestinationType:       job.Upload.DestinationType,
+					})
+				}
+
+				successfulStagingFileIDs = append(successfulStagingFileIDs, resp.JobID)
+			}
+
+			err = lf.LoadRepo.Insert(ctx, loadFiles)
+			if err != nil {
+				return fmt.Errorf("inserting load files: %w", err)
+			}
+			err = lf.StageRepo.SetStatuses(ctx, successfulStagingFileIDs, warehouseutils.StagingFileSucceededState)
+			if err != nil {
+				return fmt.Errorf("setting staging file status to succeeded: %w", err)
+			}
+			return nil
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return 0, 0, err
+	}
+
+	loadFiles, err := lf.LoadRepo.GetByStagingFiles(ctx, repo.StagingFileIDs(toProcessStagingFiles))
+	if err != nil {
+		return 0, 0, fmt.Errorf("getting load files: %w", err)
+	}
+	if len(loadFiles) == 0 {
+		return 0, 0, fmt.Errorf(`no load files generated. Sample error: %v`, sampleError)
+	}
+
+	if !slices.IsSortedFunc(loadFiles, func(a, b model.LoadFile) bool {
+		return a.ID < b.ID
+	}) {
+		return 0, 0, fmt.Errorf(`assertion: load files returned from repo not sorted by id`)
+	}
+
+	// verify if all load files are in same folder in object storage
+	if slices.Contains(warehousesToVerifyLoadFilesFolder, job.Warehouse.Type) {
+		for _, loadFile := range loadFiles {
+			if !strings.Contains(loadFile.Location, uniqueLoadGenID) {
+				err = fmt.Errorf(`all loadfiles do not contain the same uniqueLoadGenID: %s`, uniqueLoadGenID)
+				return 0, 0, err
+			}
+		}
+	}
+
+	return loadFiles[0].ID, loadFiles[len(loadFiles)-1].ID, nil
+}
+
+func (lf *LoadFileGenerator) destinationRevisionIDMap(ctx context.Context, job model.UploadJob) (revisionIDMap map[string]backendconfig.DestinationT, err error) {
+	revisionIDMap = make(map[string]backendconfig.DestinationT)
+
+	// TODO: ensure DestinationRevisionID is populated
+	for _, file := range job.StagingFiles {
+		revisionID := file.DestinationRevisionID
+		// No need to make config backend api call for the current config
+		if revisionID == job.Warehouse.Destination.RevisionID {
+			revisionIDMap[revisionID] = job.Warehouse.Destination
+			continue
+		}
+
+		destination, err := lf.ControlPlaneClient.DestinationHistory(ctx, revisionID)
```

Comment:
It is a good idea, but this is a part of the code that rarely gets hit.

For this code to get executed you need to change the destination configuration while staging files are already here. The next sync, upload will hit the previous revision only once to fetch the staging files. 

Thus the cache will not get utilised. As I believe it would be unlikely for a request with the same revision id to happen again.